### PR TITLE
imageproxy: Add a method to get image configuration

### PR DESCRIPTION
**Pull request by cgwalters**
We need this for a few reasons:

 - It has the layer ordering, which we are *currently* ignoring
   but will probably be important for correctness later
 - It is the only thing that has extensible metadata that will
   survive round trips through docker schema 2

### CODE REVIEW

**Comment by cgwalters** [comment]
Depends https://github.com/containers/skopeo/pull/1523

**Review by jlebon — APPROVED** [review]
Looks sane to me.

**Comment by cgwalters** [comment]
OK, I've updated this :new: with infrastructure to support runtime matching of the semver, moving new API methods to a new struct.

**Comment by cgwalters** [comment]
Tested in concert with https://github.com/ostreedev/ostree-rs-ext/pull/176

**Comment by cgwalters** [comment]
My OCD didn't allow having a runtime `unwrap()` on the `Option`, I realized that it's actually more elegant to represent the transient state of "proxy created but haven't fetched semver yet" by representing it as a version of 0.0.0.  Then there's no need for a potential runtime panic.

**Review by jlebon — APPROVED** [review]
LGTM!

File: src/imageproxy.rs [hunk 1]
```diff
@@ -220,10 +226,20 @@ impl ImageProxy {
                 supported
             ));
         }
+        r.protover = protover;
 
         Ok(r)
     }
 
+    /// If the proxy supports at least semver 0.2.3, return a wrapper with methods from that version.
```
Comment (jlebon) [inline src/imageproxy.rs:234]: Interesting approach. I can imagine this becoming unwieldy if we have lots of different versions we need to expose. But I guess clients are coupled enough that we can ensure there's not a large skew to maintain.

**Pull request merged**